config/cht/test/contact-summary/active_pregnancy.spec.js

```diff
@@ -0,0 +1,287 @@
+    if (clock) clock.restore();
+  });
+
+  it('pregnancy registration with risk factors, danger signs, past and future visits', async () => {
+    await harness.setNow('2000-01-01');
+
+    clock = sinon.useFakeTimers(moment('2000-01-01').toDate());
+    // Load the pregnancy form and fill in
+    const result = await harness.fillForm('pregnancy', ...pregnancyRegistrationScenarios.riskDanger);
+
+    // Verify that the form successfully got submitted
+    expect(result.errors).to.be.empty;
+
+    // Verify condition card
+    const contactSummary = Object.assign({}, harness.content, harness.getContactSummary());
+    expect(contactSummary.cards).to.have.property('length', 1);
+    const activePregnancyCard = contactSummary.cards[0];
+    expect(activePregnancyCard).to.have.property('label', 'contact.profile.pregnancy.active');
+    const fields = activePregnancyCard.fields;
+    expect(fields).to.have.property('length', 7);
+    expect(fields[0]).to.deep.equal(
+      {
+        "label": "Weeks Pregnant",
+        "value": {
+          "number": 21,//21 weeks and 6 days
+          "approximate": false
+        },
+        "translate": false,
+        "filter": "weeksPregnant",
+        "width": 6
+      });
+    expect(fields[1]).to.deep.equal({
+      "label": "contact.profile.edd",
+      "value": 957636900000,//2000-05-07 (LMP date + 280 days)
+      "translate": false,
+      "filter": "simpleDate",
+      "width": 6
+    });
+    expect(fields[2]).to.deep.equal({
+      "label": "contact.profile.risk.high",
+      "value": "Asthma; underweight",
+      "icon": "icon-risk",
+      "width": 6
+    });
+    expect(fields[3]).to.deep.equal({
+      "label": "contact.profile.danger_signs.current",
+      "value": "Vaginal bleeding",
+      "width": 6
+    });
+    expect(fields[4]).to.deep.equal({
+      "label": "contact.profile.visit",
+      "value": "contact.profile.visits.of",
+      "context": {
+        "count": 1,
+        "total": 8
+      },
+      "translate": true,
+      "width": 6
+    });
+    expect(fields[5]).to.deep.equal({
+      "label": "contact.profile.last_visited",
+      "value": "0 weeks ago",
+      "width": 6
+    });
+    expect(fields[6]).to.deep.equal({
+      "label": "contact.profile.anc.next",
+      "value": 947873700000,//2000-01-15
+      "filter": "simpleDate",
+      "width": 6
+    });
+  });
+
+  it('pregnancy registration with no known LMP, no risk factors, no danger signs, no past and future viits', async () => {
+    await harness.setNow('2000-01-01');
+
+    clock = sinon.useFakeTimers(moment('2000-01-01').toDate());
+    // Load the pregnancy form and fill in
+    const result = await harness.fillForm('pregnancy', ...pregnancyRegistrationScenarios.lmpUnknown);
+
+    // Verify that the form successfully got submitted
+    expect(result.errors).to.be.empty;
+
+    // Verify condition card
+    const contactSummary = Object.assign({}, harness.content, harness.getContactSummary());
```

Comment:
Can you explain why you combine harness.content with the contactSummary instead of asserting on their individual attributes?

Reply:
I had tried various ways of combining `harness.content` with contact-summary, and it should have stuck on this. It needs to be cleaned. Will do it. Thanks.

Reply:
Done.